Broken Webpack compatibility

PR https://github.com/polkadot-js/api/pull/1758

This PR broke ability to bundle the api package with wepback because of unsafe require.

I used to bundle this package with webpack, but updating to the new version (from v0.100.1 to v1.0.1) broke webpack bundle ability.

My webpack.config.js:
```javascript
        target: "node",
        mode: "development",
        output: {
            path: path.resolve(profilePath, 'dist'),
            filename: 'file.js',
            libraryTarget: 'umd',
            library: 'profile',
            umdNamedDefine: true
        },
        resolve: {
            extensions: ['.ts', '.js']
        },
        node: {
            fs: "empty"
        },
        module: {
            exprContextCritical: false,
            rules: [
                {
                    test: /\.ts$/,
                    loader: require.resolve('ts-loader'),
                    exclude: /node_modules/,
                    options: {
                        configFile: tsconfig,
                        context: profilePath,
                    }
                }
            ]
        }
```

Webpack error:
`
./node_modules/@polkadot/api/index.js
Module not found: Error: Can't resolve '../package.json' in '/myproject/node_modules/@polkadot/api'
resolve '../package.json' in '/myproject/node_modules/@polkadot/api'
  using description file: /myproject/node_modules/@polkadot/api/package.json (relative path: .)
    using description file: /myproject/package.json (relative path: ./node_modules/@polkadot/package.json)
      no extension
        /myproject/node_modules/@polkadot/package.json doesn't exist
      .ts
        /myproject/node_modules/@polkadot/package.json.ts doesn't exist
      .js
        /myproject/node_modules/@polkadot/package.json.js doesn't exist
      as directory
        /myproject/node_modules/@polkadot/package.json doesn't exist
[/myproject/node_modules/@polkadot/package.json]
[/myproject/node_modules/@polkadot/package.json.ts]
[/myproject/node_modules/@polkadot/package.json.js]
`

--- a/jest.config.js
+++ b/jest.config.js
@@ -31,5 +31,6 @@ module.exports = Object.assign({}, config, {
     ws_local_polkadot_alex: 'ws://127.0.0.1:9949/',
     ws_substrate_1: 'wss://substrate-rpc.parity.io/',
     ws_polkadot_alex: 'wss://poc3-rpc.polkadot.io/'
-  }
+  },
+  resolver: './jest.resolver.js'
 });

--- a/jest.resolver.js
+++ b/jest.resolver.js
@@ -1,0 +1,15 @@
+// Copyright 2017-2020 @polkadot/util authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const path = require('path');
+
+module.exports = function resolver (file, config) {
+  if (file.includes('package.json')) {
+    console.error(file, config);
+    return path.join(config.basedir.replace('/src', '/'), file);
+  }
+
+  return config.defaultResolver(file, config);
+};

--- a/packages/api-derive/package.json
+++ b/packages/api-derive/package.json
@@ -32,6 +32,6 @@
     "@polkadot/types": "^1.1.0-beta.21"
   },
   "devDependencies": {
-    "@polkadot/keyring": "^2.1.1"
+    "@polkadot/keyring": "^2.2.0-beta.0"
   }
 }

--- a/packages/api/package.json
+++ b/packages/api/package.json
@@ -25,20 +25,17 @@
     "url": "https://github.com/polkadot-js/api/issues"
   },
   "homepage": "https://github.com/polkadot-js/api/tree/master/packages/api#readme",
-  "browser": {
-    "../package.json": "./package.json"
-  },
   "dependencies": {
     "@babel/runtime": "^7.8.3",
     "@polkadot/api-derive": "^1.1.0-beta.21",
-    "@polkadot/keyring": "^2.1.1",
+    "@polkadot/keyring": "^2.2.0-beta.0",
     "@polkadot/metadata": "^1.1.0-beta.21",
     "@polkadot/rpc-core": "^1.1.0-beta.21",
     "@polkadot/rpc-provider": "^1.1.0-beta.21",
     "@polkadot/types": "^1.1.0-beta.21",
-    "@polkadot/util-crypto": "^2.1.1"
+    "@polkadot/util-crypto": "^2.2.0-beta.0"
   },
   "devDependencies": {
-    "@polkadot/keyring": "^2.1.1"
+    "@polkadot/keyring": "^2.2.0-beta.0"
   }
 }

--- a/packages/api/src/index.ts
+++ b/packages/api/src/index.ts
@@ -4,22 +4,13 @@
 
 import { detectPackage } from '@polkadot/util';
 
-let dirname;
-let pkgJson;
-
+// eslint-disable-next-line no-useless-catch
 try {
-  dirname = __dirname;
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  detectPackage(require('./package.json'), __dirname);
 } catch (error) {
-  // ignore
+  throw error;
 }
-
-try {
-  pkgJson = require('./package.json');
-} catch (error) {
-  pkgJson = require('../package.json');
-}
-
-detectPackage(pkgJson, dirname);
 
 export { Keyring } from '@polkadot/keyring';
 export { WsProvider } from '@polkadot/rpc-provider';

--- a/packages/metadata/package.json
+++ b/packages/metadata/package.json
@@ -28,10 +28,10 @@
   "dependencies": {
     "@babel/runtime": "^7.8.3",
     "@polkadot/types": "^1.1.0-beta.21",
-    "@polkadot/util": "^2.1.1",
-    "@polkadot/util-crypto": "^2.1.1"
+    "@polkadot/util": "^2.2.0-beta.0",
+    "@polkadot/util-crypto": "^2.2.0-beta.0"
   },
   "devDependencies": {
-    "@polkadot/keyring": "^2.1.1"
+    "@polkadot/keyring": "^2.2.0-beta.0"
   }
 }

--- a/packages/rpc-core/package.json
+++ b/packages/rpc-core/package.json
@@ -30,7 +30,7 @@
     "@polkadot/jsonrpc": "^1.1.0-beta.21",
     "@polkadot/rpc-provider": "^1.1.0-beta.21",
     "@polkadot/types": "^1.1.0-beta.21",
-    "@polkadot/util": "^2.1.1",
+    "@polkadot/util": "^2.2.0-beta.0",
     "rxjs": "^6.5.4"
   }
 }

--- a/packages/rpc-provider/package.json
+++ b/packages/rpc-provider/package.json
@@ -28,14 +28,14 @@
   "dependencies": {
     "@babel/runtime": "^7.8.3",
     "@polkadot/metadata": "^1.1.0-beta.21",
-    "@polkadot/util": "^2.1.1",
-    "@polkadot/util-crypto": "^2.1.1",
+    "@polkadot/util": "^2.2.0-beta.0",
+    "@polkadot/util-crypto": "^2.2.0-beta.0",
     "eventemitter3": "^4.0.0",
     "isomorphic-fetch": "^2.2.1",
     "websocket": "^1.0.31"
   },
   "devDependencies": {
-    "@polkadot/keyring": "^2.1.1",
+    "@polkadot/keyring": "^2.2.0-beta.0",
     "mock-socket": "^9.0.3",
     "nock": "^11.7.2"
   }

--- a/packages/types/package.json
+++ b/packages/types/package.json
@@ -35,13 +35,13 @@
   "dependencies": {
     "@babel/runtime": "^7.8.3",
     "@polkadot/metadata": "^1.1.0-beta.21",
-    "@polkadot/util": "^2.1.1",
-    "@polkadot/util-crypto": "^2.1.1",
+    "@polkadot/util": "^2.2.0-beta.0",
+    "@polkadot/util-crypto": "^2.2.0-beta.0",
     "memoizee": "^0.4.14",
     "yargs": "^15.1.0"
   },
   "devDependencies": {
-    "@polkadot/keyring": "^2.1.1",
+    "@polkadot/keyring": "^2.2.0-beta.0",
     "@types/memoizee": "^0.4.3",
     "@types/yargs": "^15.0.2"
   }

